test/gql_spec.js

```diff
@@ -1,20 +1,606 @@
+        });
+
+        it('should support greater than matches', function () {
+            gql.parse({created_at: {$gt: '2016-03-02'}}).conditions
+                .should.eql({where: ['created_at', '>', '2016-03-02']});
+            gql.parse('created_at:>\'2016-03-02\'').conditions
+                .should.eql({where: ['created_at', '>', '2016-03-02']});
+        });
+
+        it('should support not greater than matches', function () {
+            gql.parse({created_at: {$not: {$gt: '2016-03-02'}}}).conditions
+                .should.eql({whereNot: ['created_at', '>', '2016-03-02']});
+            gql.parse('!created_at:>\'2016-03-02\'').conditions
+                .should.eql({whereNot: ['created_at', '>', '2016-03-02']});
+        });
+
+        it('should support less than equal matches', function () {
+            gql.parse({created_at: {$lte: '2016-03-02'}}).conditions
+                .should.eql({where: ['created_at', '<=', '2016-03-02']});
+            gql.parse('created_at:<=\'2016-03-02\'').conditions
+                .should.eql({where: ['created_at', '<=', '2016-03-02']});
+        });
+
+        it('should support not less than equal matches', function () {
+            gql.parse({created_at: {$not: {$lte: '2016-03-02'}}}).conditions
+                .should.eql({whereNot: ['created_at', '<=', '2016-03-02']});
+            gql.parse('!created_at:<=\'2016-03-02\'').conditions
+                .should.eql({whereNot: ['created_at', '<=', '2016-03-02']});
+        });
+
+        it('should support greater than equal matches', function () {
+            gql.parse({created_at: {$gte: '2016-03-02'}}).conditions
+                .should.eql({where: ['created_at', '>=', '2016-03-02']});
+            gql.parse('created_at:>=\'2016-03-02\'').conditions
+                .should.eql({where: ['created_at', '>=', '2016-03-02']});
+        });
+
+        it('should support not greater than equal matches', function () {
+            gql.parse({created_at: {$not: {$gte: '2016-03-02'}}}).conditions
+                .should.eql({whereNot: ['created_at', '>=', '2016-03-02']});
+            gql.parse('!created_at:>=\'2016-03-02\'').conditions
+                .should.eql({whereNot: ['created_at', '>=', '2016-03-02']});
+        });
+
+        it('should support null matches', function () {
+            gql.parse({created_at: null}).conditions
+                .should.eql({whereNull: ['created_at']});
+            gql.parse('created_at:null').conditions
+                .should.eql({whereNull: ['created_at']});
+        });
+
+        it('should support not null matches', function () {
+            gql.parse({$not: {created_at: null}}).conditions
+                .should.eql({whereNotNull: ['created_at']});
+            gql.parse('!created_at:null').conditions
+                .should.eql({whereNotNull: ['created_at']});
+        });
+
+        it('should support Javascript Date values through the JSON API', function (done) {
```

Comment:
This is somewhat relevant to #6. It's a different approach for sure. Actual date objects are only supported in the JSON API. But, it would be possible to add something into GQL to interpret property values during `.parse()`. That should provide the support that you're really looking for in #6.